### ISSUE TITLE
A Suspected Bug in binary_crossentropy

### ISSUE THREAD

**Issue by River861** [tensorflow/tensorflow#51675]
Hi, I found that the the calculation result of `binary_crossentropy` loss function is different from other deep learning libraries, such as theano.

Here is an example code using tensorflow2.4.3:
```python
import numpy as np
import tensorflow as tf

y_true = np.array([0., 1., 0.], dtype=np.float32)
y_pred = np.array([0.9999999, 0.9999999, 0.0000001], dtype=np.float32)

res = tf.keras.backend.binary_crossentropy(y_true, y_pred)

print("loss = ", res.numpy())
```
The result is:
```
loss =  [15.333239 -0.       -0.      ]
```

And this is the code using theano1.0.4:
```python
import numpy as np
from theano import tensor as T

y_true = np.array([0., 1., 0.], dtype=np.float32)
y_pred = np.array([0.9999999, 0.9999999, 0.0000001], dtype=np.float32)

res = T.nnet.binary_crossentropy(y_pred, y_true)

print("loss = ", res.eval())
```
The result is:
```
loss =  [1.5942385e+01 1.1920930e-07 1.1920930e-07]
```

I then found the cause of the inconsistency is that, tensorflow use the `epsilon`  to caculate the loss value,  which I think is redundant because the output has been clipped using `epsilon` eariler. Here is the source code location:
https://github.com/tensorflow/tensorflow/blob/fbd7286aba58ba180a2e3c8a280ed5379ee5435d/tensorflow/python/keras/backend.py#L5047-L5052

Besides, I found that the `categorical_crossentropy` doesn't use the `epsilon` to caculate the loss value. This makes me more suspicious of the usage of `epsilon` in `binary_crossentropy`:
https://github.com/tensorflow/tensorflow/blob/fbd7286aba58ba180a2e3c8a280ed5379ee5435d/tensorflow/python/keras/backend.py#L4906-L4908

Thanks.
